uni/lib/controller/networking/network_router.dart

```diff
@@ -89,29 +92,140 @@ class NetworkRouter {
     );
   }
 
+  static Future<Session?> loginWithToken(
+    String token,
+    String studentNumber,
+    List<String> faculties, {
+    required bool persistentSession,
+  }) async {
```

Comment:
This function is a bit too big. Can we put it in a separate file with helpers?

Reply:
I am not a big fan of helpers and this function is right after the other login method